File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        int mid = (left + right) / 2;
        // loop while left < right;
        while(left <= right){           
            cout << nums[mid] << endl;
            // else if target is less than nums [mid] then right is nums[mid - 1]
            if(target == nums[mid]){
                return mid;
            } else if (target > nums[mid]){
                left = mid + 1;
                mid = (left + right) / 2;
            } else{
                right = mid - 1;
                mid = (left + right) / 2;
            }
        }
        return -1;
    }
};

// Divide and conquer
// use two pointer and a middle var

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        if len(nums) < 2:
            return False

        dictionary = {}

        for i, value in enumerate(nums):
            complement = target - value
            if complement in dictionary:
                return [dictionary[complement], i]
            else:
                dictionary[value] = i

        return False